src/middlewared/middlewared/plugins/filesystem.py

```diff
@@ -285,6 +289,19 @@ def __convert_to_adv_flagset(self, basic_flag):
 
         return flagset
 
+    @private
+    @job(lock=lambda args: f'_winacl:{args[0]}')
+    def _winacl(self, job, path, action, uid, gid, options):
```

Comment:
Why have you decided to make it a job?

Reply:
This can potentially take several hours to complete if there are a significant number of files involved.

Reply:
So we are calling it asynchronously them, `filesystem.acl` returns success immediately and if any exception happens in `_winacl`, it is never shown anywhere?